lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake

```diff
@@ -1,6 +1,13 @@
 namespace :katello do
   namespace :upgrades do
     namespace '3.11' do
+      def wait_on_task(task)
+        task = SmartProxy.pulp_master.pulp_api.resources.task.poll(task['task_id'])
+        return if Actions::Pulp::AbstractAsyncTask::FINISHED_STATES.include?(task['state'])
+        sleep 1
```

Comment:
could this cause *really* long upgrades if there are thousands of distributors to process and every single call needs a sleep 1? 

perhaps a shorter sleep time if we have to rely on this mechanism (0.1)..

Reply:
we could do a shorter mechanism or have it increase, i'll update to 0.1 seconds and then increase by 0.2 seconds every iteration

Reply:
done